tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs

```diff
@@ -68,36 +79,64 @@ public void CopyTo(IConfigurationTelemetry destination)
                 return data;
             }
 
-            while (config.TryDequeue(out var entry))
-            {
-                data.Add(GetConfigKeyValue(entry));
-            }
+            GetData(config, data);
 
             return data;
+        }
 
-            static ConfigurationKeyValue GetConfigKeyValue(ConfigurationTelemetryEntry entry)
+        private static void GetData(ConcurrentQueue<ConfigurationTelemetryEntry> buffer, List<ConfigurationKeyValue> destination)
+        {
+            while (buffer.TryDequeue(out var entry))
             {
-                return new ConfigurationKeyValue(
-                    name: entry.Key,
-                    origin: entry.Origin.ToStringFast(),
-                    seqId: entry.SeqId,
-                    error: entry.Error,
-                    value: GetValue(entry));
-            }
+                destination.Add(GetConfigKeyValue(entry));
 
-            static object? GetValue(ConfigurationTelemetryEntry entry)
-            {
-                return entry.Type switch
+                if (ConfigurationKeysMapping.TryGetValue(entry.Key, out var mappedKey))
                 {
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue,
-                    ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
-                    _ => entry.StringValue
-                };
+                    destination.Add(MapConfigKeyValueAsString(entry, mappedKey));
+                }
             }
         }
 
+        private static object? GetValue(ConfigurationTelemetryEntry entry)
+        {
+            return entry.Type switch
+            {
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Double => entry.DoubleValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Int => entry.IntValue,
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Redacted => "<redacted>",
+                _ => entry.StringValue
+            };
+        }
+
+        private static string? GetValueAsString(ConfigurationTelemetryEntry entry)
+        {
+            return entry.Type switch
+            {
+                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue?.ToString()?.ToLowerInvariant(),
```

Comment:
Isn't this easier/faster/cheaper?

```suggestion
                ConfigurationTelemetry.ConfigurationTelemetryEntryType.Bool => entry.BoolValue ? "true" : "false",
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5126/commits/7d4b6a84d089ba85674e50047ce60bc18a403cfc